ic-agent/src/agent/mod.rs

```diff
@@ -330,12 +330,15 @@ impl Agent {
     }
 
     fn get_expiry_date(&self) -> u64 {
-        // TODO(hansl): evaluate if we need this on the agent side (my hunch is we don't).
-        let permitted_drift = Duration::from_secs(60);
-        self.ingress_expiry
-            .as_nanos()
-            .saturating_add(OffsetDateTime::now_utc().unix_timestamp_nanos() as u128)
-            .saturating_sub(permitted_drift.as_nanos()) as u64
+        let expiry_raw = OffsetDateTime::now_utc() + self.ingress_expiry;
+        let mut rounded = expiry_raw.replace_nanosecond(0).unwrap();
+        if self.ingress_expiry.as_secs() > 60 {
```

Comment:
Why is rounding only done if ingress_expiry is larger than 60 seconds? 
Is this aligned between the BN team (caching), agent-js and agent-rs?